### DISCUSSION TITLE
Access nested variable in pre-request script

### DISCUSSION THREAD

**Discussion by mr-nothing** [Kong/insomnia#7833] [Help] [answered]
Hey guys!

I'm migrating from Postman to Insomnia and struggling to access nested variable in my pre-request script that does authorisation before every request.
The think is that this is working `console.log(insomnia.variables.get('authentication'));` but these options don't work:
```
console.log(insomnia.variables.get('{{ _.service.auth.legacy }}'));
console.log(insomnia.variables.get(service.auth.legacy));
console.log(insomnia.variables.get(_.service.auth.legacy));
```

What am I doing wrong? Is this possible?

**Comment by ihexxa** [comment, marked as answer]
Hey,
Hm I don't see a way to access nested value directly in Postman. In Insomnia, it intentionally provides a similar way for accessing values, additionally it is able to return an object instead of string, e.g:

```javascript
const parent = insomnia.variables.get('parent'); // parent is an object
console.log(parent.property);
```

**Reply by mr-nothing** [reply]
Thank you so much for your help! Never used this feature so this is fine, don't think anything need to be changed.
